### ISSUE TITLE
Compatible for an iOS 8 Cocoa Touch Framework

### ISSUE THREAD

**Issue by iblacksun** [abbeycode/UnzipKit#21]
A similar discussion for the problem can be found here: https://github.com/ZipArchive/ZipArchive/issues/158

**Comment by abbeycode** [comment]
I'll take a look and let you know what I find. Were you getting "Include of non-modular header inside framework module"?

**Comment by abbeycode** [comment]
I created a test project in Xcode 7.0.1, targeting iOS 8.4, using the latest CocoaPods release of UnzipKit (v1.4.2) and didn't get any compiler errors (just one warning).

**Comment by iblacksun** [comment]
Thanks for your response, here is a demo project https://github.com/iblacksun/UnzipKitIssue

I want to use UnzipKit in a Swift framework.

**Comment by abbeycode** [comment]
It looks like this was already fixed in the Swift-2.0 branch. I'll update this thread once I cut a release with those changes, but for now, this should work in your Podfile:

    pod 'UnzipKit', :git => 'https://github.com/abbeycode/UnzipKit.git', :branch => 'swift-2.0'

**Comment by iblacksun** [comment]
I'm using ```pod 'UnzipKit', :git => 'https://github.com/abbeycode/UnzipKit.git', :branch => 'swift-2.0'````, but issue still exist.

**Comment by abbeycode** [comment]
Did you do a `pod update` and then clean your build?

**Comment by iblacksun** [comment]
Sure, 
1. Remove this pod from Pod file.
2. Do a ```pod update```.
3. Add pod with swift-2.0 branch.
3. Do a ```pod update``` again, and clean project.

**Comment by abbeycode** [comment]
Sorry, my mistake. It started building a different scheme after I updated. I'll have to take another look later. Thanks for giving me the sample project though, it should make it a lot easier.

**Comment by abbeycode** [comment]
According to someone from CocoaPods in the issue above, it doesn't support linking to framework targets at this time. Could you link UnzipKit to your app target instead? If not, Carthage should be at least mostly supported. I have a branch ([`carthage`](https://github.com/abbeycode/UnzipKit/tree/carthage)) that should bring 100% compatibility.

**Comment by iblacksun** [comment]
Thank you, i will try git submodule or carthage.

**Comment by iblacksun** [comment]
Sorry, once again.
I 'm using Carthage, if i use ```carthage``` branch, ```carthage update``` will failure.
```swift-2.0``` branch is ok, but after ```carthage update```, in build directory only one Mac framework.

**Comment by abbeycode** [comment]
Did it still not work outside of CocoaPods?

**Comment by abbeycode** [comment]
Ah, I see. I'll merge the `swift-2.0` changes into `carthage`

**Comment by iblacksun** [comment]
Sorry, BUILD FAILED again, This is log below.

```
*** Fetching UnzipKit
*** Checking out UnzipKit at "bb05df91f0040e8257521853b4f31ed791fca27d"
*** xcodebuild output can be found in /var/folders/q9/btclmwn172g5pr4qgm_b4ms00000gn/T/carthage-xcodebuild.zMmuds.log
*** Building scheme "UnzipKit" in UnzipKit.xcworkspace
*** Building scheme "UnzipKit-iOS" in UnzipKit.xcworkspace
** BUILD FAILED **


The following build commands failed:
	CompileC /Users/iblacksun/Library/Developer/Xcode/DerivedData/UnzipKit-dsjmzyuntkxtcxalqdwlvbzbfwoq/Build/Intermediates/UnzipKit.build/Release-iphonesimulator/UnzipKit-iOS.build/Objects-normal/x86_64/UZKArchive.o Source/UZKArchive.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
	CompileC /Users/iblacksun/Library/Developer/Xcode/DerivedData/UnzipKit-dsjmzyuntkxtcxalqdwlvbzbfwoq/Build/Intermediates/UnzipKit.build/Release-iphonesimulator/UnzipKit-iOS.build/Objects-normal/i386/UZKArchive.o Source/UZKArchive.m normal i386 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(2 failures)
```

**Comment by abbeycode** [comment]
I apologize for the delay (I had a family emergency come up). Can you please try again? I've made some fixes, and `carthage update` succeeds with this Cartfile:

```
github "abbeycode/UnzipKit" "carthage"
```

**Comment by iblacksun** [comment]
Thanks very much.

Now UnzipKit can build success with carthage, but when using in framework, Xcode still report errors.

I have update the demo project, could you check it for me?

https://github.com/iblacksun/UnzipKitIssue

**Comment by abbeycode** [comment]
I poked around, and tried a bunch of different solutions, none of which worked. I posted about it [on StackOverflow](http://stackoverflow.com/q/33438654/105717), so let's see if we get any traction there.

**Comment by iblacksun** [comment]
Thanks very much.
I had contact the Apple Developer Technical Support to request help. when they reply me or have a solution, i will update here.

**Comment by abbeycode** [comment]
FYI, I merged all branches to trunk and cut a new CocoaPods release.

**Comment by iblacksun** [comment]
DTS said:

DTS supports developers using Apple-authored tools and frameworks only. We are unable to provide support for third-party tools, development environments, tutorials or other resources.

**Comment by abbeycode** [comment]
This should be fixed as of v1.6. As mentioned above, use Carthage (on `master` now) and you should be fine.
